tests/test_eval_detection.py

```diff
@@ -0,0 +1,147 @@
+# <Copyright 2020, Argo AI, LLC. Released under the MIT license.>
+"""Detection evaluation unit tests"""
+
+import logging
+import pathlib
+from typing import List
+
+import numpy as np
+import pytest
+from pandas.core.frame import DataFrame
+from scipy.spatial.transform import Rotation as R
+
+from argoverse.data_loading.object_label_record import ObjectLabelRecord
+from argoverse.evaluation.detection_utils import AffFnType, DistFnType, compute_affinity_matrix, dist_fn, iou_aligned_3d
+from argoverse.evaluation.eval_detection import SIGNIFICANT_DIGITS, DetectionCfg, DetectionEvaluator
+from argoverse.utils.transform import quat_scipy2argo_vectorized
+
+TEST_DATA_LOC = pathlib.Path(__file__).parent.parent / "tests" / "test_data" / "detection"
+logging.getLogger("matplotlib.font_manager").disabled = True
+
+
+@pytest.fixture
+def evaluator() -> DetectionEvaluator:
+    """Define an evaluator that compares a set of results to itself."""
+    detection_cfg = DetectionCfg(detection_classes=["VEHICLE"])
+    return DetectionEvaluator(
+        TEST_DATA_LOC / "detections", TEST_DATA_LOC, TEST_DATA_LOC / "test_figures", detection_cfg
+    )
+
+
+@pytest.fixture
+def metrics(evaluator: DetectionEvaluator) -> DataFrame:
+    return evaluator.evaluate()
+
+
+def test_affinity_center() -> None:
+    """Intialize a detection and a ground truth label. Verify that calculated distance matches expected affinity
+    under the specified `AffFnType`.
+    """
+    dts: List[ObjectLabelRecord] = [ObjectLabelRecord(np.array([0, 0, 0, 0]), np.array([0, 0, 0]), 5.0, 5.0, 5.0, 0)]
```

Comment:
might be a bit cleaner to write the following so the reader doesn't have to guess the arguments here
```python
    dts: List[ObjectLabelRecord] = [
        ObjectLabelRecord(
             quaternion=np.array([0, 0, 0, 0]), 
             translation=np.array([0, 0, 0]),
             length=5.0,
             width=5.0,
             height=5.0,
             occlusion=0
       )
   ]

        self.length = length
        self.width = width
        self.height = height
        self.occlusion = occlusion
        self.label_class = label_class
        self.track_id = track_id
```